### PR TITLE
Scheduled toot

### PR DESCRIPTION
**Pull request by kyori19**
Related to #981 .

I added button to configure scheduled status and add `scheduled_at` param to `createStatus` method.

But there seems to be a problem with scheduled toots contains images.(tested on Mastodon v2.7.0 master)
This is not a bug of this PR.

My toot posted with this feature. -> https://odakyu.app/@ars42525/101455089752932693

### CODE REVIEW

**Comment by charlag** [comment]
Woah, thanks, that would be super cool to have!
My biggest concern so far is UI-wise. I'll try to go through the code during the week.

File: app/src/main/java/com/keylesspalace/tusky/fragment/DatePickerFragment.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import java.util.Calendar;
+import java.util.TimeZone;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.DialogFragment;
+
+public class DatePickerFragment extends DialogFragment {
+
+    public static final String PICKER_TAG = "date_picker";
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/fragment/DatePickerFragment.java]: Do we need this tag?

File: app/src/main/java/com/keylesspalace/tusky/network/MastodonApi.java [hunk 2]
```diff
@@ -33,6 +31,7 @@
 
 import java.util.List;
 
+import androidx.annotation.Nullable;
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/network/MastodonApi.java]: Please remove unused imports in few places

Comment (kyori19) [reply app/src/main/java/com/keylesspalace/tusky/network/MastodonApi.java]: I think there's no unused imports in this PR.
Am I missing something?

**Comment by kyori19** [comment]
* Moved `PICKER_TAG` to `picker.show(fragmentManager, tag)`
* Added `VersionUtils` to check Mastodon version and hide schedule button if it is lower than 2.7.0

I'm not good at writing Kotlin. Can I use Android Studio's "migrate to Kotlin" feature?

File: app/src/main/java/com/keylesspalace/tusky/network/MastodonApi.java (hunk 2 above)
Comment (charlag) [reply app/src/main/java/com/keylesspalace/tusky/network/MastodonApi.java]: Maybe I'm reading incorrectly on Github, sorry

**Comment by charlag** [comment]
I usually convert existing things automatically and then fix rough edges. We can try to do this if you want but if you would rather not then just leave it like that.

**Comment by connyduck** [comment]
- The date/time picker dialog is ugly as hell, because its not from the material design library. Coincidentally, the Android team just started to work on one. I guess we have to wait until they release it. https://github.com/material-components/material-components-android/blob/704aff2f14d3b985ab35fe83e22d239c08caf943/lib/java/com/google/android/material/picker/MaterialDatePickerDialog.java

- Nice idea with the bottom sheet. I would change the behavior so that it hides after a time was selected, but the clock icon turns blue to indicate the toot will be sent scheduled.

- There should be an icon next to the set time/date to indicate its editable.

- There has to be a way to remove the time and send the status immediately.

**Comment by kyori19** [comment]
I found problem with timeline auto-reloading after toot so I fixed it and rebased branch.

**Comment by charlag** [comment]
@kyori19 sorry that it takes too long, we'll try to get back to it whenever possible!
You need a new review at this stage, right?

**Comment by kyori19** [comment]
@charlag 
Thanks, I'm sure I need some reviews to improve this feature. And I have to make change the UI, so I will work on it in near.

**Comment by kyori19** [comment]
I improved bottom sheet design.

Added edit button, added reset button and schedule button turns blue when a time is set.

**Comment by charlag** [comment]
I would revese right and left part of the sheet and maybe made reset button less prominent.
Edit icon doesn't seem aligned with text. But nice progress!
(I'm not maintainer and I just leave my feedback, feel free to ignore it)

**Comment by kyori19** [comment]
@charlag Thanks!
I will fix about the edit icons alignment sooner.
I think it is better that the edit icons are placed on the right side of text.
If the reset button placed on the right of the sheet, strange space will appear on left of the edit icon.
But I'm also one of contributor, so I want to hear @connyduck's opinion.

**Comment by kyori19** [comment]
Fixed edit icon's size.
image

**Comment by kyori19** [comment]
Added a list of scheduled toots to drawer.
There's no API to edit scheduled status so once delete it and show `ComposeActivity` when edit statuses.

image image

File: app/src/main/java/com/keylesspalace/tusky/ComposeActivity.java [hunk 3]
```diff
@@ -1152,7 +1196,14 @@ private void addMediaToQueue(@Nullable String id, QueuedMedia.Type type, Bitmap
         layoutParams.setMargins(margin, 0, margin, marginBottom);
         view.setLayoutParams(layoutParams);
         view.setScaleType(ImageView.ScaleType.CENTER_CROP);
-        view.setImageBitmap(preview);
+        if (preview != null) {
+            view.setImageBitmap(preview);
+        } else {
+            Picasso.with(this)
```
Comment (kyori19) [inline app/src/main/java/com/keylesspalace/tusky/ComposeActivity.java]: Use picasso to show media preview when it has been already uploaded.

**Comment by charlag** [comment]
@kyori19 some things have been changed in master. Would you be willing to update it?

**Comment by kyori19** [comment]
@charlag 
Sorry for waiting you long.
Rebased so that you can merge.
Furthermore, I fixed wrong version detection. 

And I am still waiting for completion of MaterialDateTimePicker so please wait...

**Comment by charlag** [comment]
Ah, sorry, I thought you've been waiting for us to merge it and it's getting older.
Good that you rebased it anyway, it's easier to do it regularly. Thanks.

**Comment by connyduck** [comment]
Looks like the newest Materialcomponents library has an updated public MaterialDatePickerDialog https://github.com/material-components/material-components-android/releases/tag/1.1.0-alpha10

Can you resolve the merge conflicts and try it out please? @kyori19 
Let me know if I can help with something

**Comment by kyori19** [comment]
@connyduck 
Yeah, I know. I already resolved all conflicts and working on implementing with new datepicker.
It seems that I need few days to complete this, so please wait...

**Comment by kyori19** [comment]
@connyduck I changed the design of ScheduleView.
I think the strange parts has been improved.
Please check them.

**Comment by kyori19** [comment]
Resolved all conflicts.

**Pull request merged**